library/aes.c

```diff
@@ -1446,44 +1446,79 @@ int mbedtls_aes_self_test( int verbose )
             mbedtls_printf( "  AES-CTR-128 (%s): ",
                              ( v == MBEDTLS_AES_DECRYPT ) ? "dec" : "enc" );
 
-        memcpy( nonce_counter, aes_test_ctr_nonce_counter[u], 16 );
         memcpy( key, aes_test_ctr_key[u], 16 );
 
-        offset = 0;
         mbedtls_aes_setkey_enc( &ctx, key, 128 );
 
         if( v == MBEDTLS_AES_DECRYPT )
         {
             len = aes_test_ctr_len[u];
-            memcpy( buf, aes_test_ctr_ct[u], len );
-
-            mbedtls_aes_crypt_ctr( &ctx, len, &offset, nonce_counter, stream_block,
-                           buf, buf );
 
-            if( memcmp( buf, aes_test_ctr_pt[u], len ) != 0 )
+            for(size_to_cipher = 0; size_to_cipher < len; size_to_cipher++ )
             {
-                if( verbose != 0 )
-                    mbedtls_printf( "failed\n" );
-
-                ret = 1;
-                goto exit;
+                memcpy( buf, aes_test_ctr_ct[u], len );
+                memcpy( nonce_counter, aes_test_ctr_nonce_counter[u], 16 );
+                offset = 0;
+
+                /* cipher all the blocks until the last one, and add the offset*/
```

Comment:
This comment is not true. For ```length == 32```, iterations ```size_to_cipher < 16``` only first block is ciphered. It may be said as "beginning block(s)" and "remaining block(s)" for next call.

Reply:
Thanks, I didn't change the comment when I updated the test